client.go

```diff
@@ -188,13 +190,16 @@ func (c *client) Scan(s *hrpc.Scan) ([]*hrpc.Result, error) {
 		res, err = c.sendRPC(rpc)
 
 		// Check to see if this region is the last we should scan (either
-		// because (1) it's the last region or (3) because its stop_key is
-		// greater than or equal to the stop_key of this scanner provided
-		// that (2) we're not trying to scan until the end of the table).
-		// (1)
-		if len(rpc.RegionStop()) == 0 ||
+		// because:
+		// (0) we got the number of rows user asked for
+		// (1) it's the last region or
+		// (3) because its stop_key is greater than or equal to the stop_key of this scanner,
+		// provided that (2) we're not trying to scan until the end of the table).
+		if uint32(len(results)) >= numberOfRows ||
```

Comment:
if this "if" statement is false, and we are to move to next region, `numberOfRows` should probably be updated to `numberOfRow - uint32(len(results))`?